common/lib/xmodule/xmodule/js/src/capa/display.coffee

```diff
@@ -157,6 +159,98 @@ class @Problem
 
   # TODO add hooks for problem types here by inspecting response.html and doing
   # stuff if a div w a class is found
+          if response.progress_status != "done"
+            @maybeSetupTimer()
+
+  getSecondsLeft: =>
+    # Returns 
+    #   a nonnegative integer representing seconds left on a timed exam.
+    #   -1 on error
+    if _.isUndefined(@startTime) or _.isUndefined(@totalSeconds)
+      return -1
+    else
+      return Math.floor(Math.max(@totalSeconds - ((new Date() - @startTime) / 1000), 0))
+
+  maybeSetupTimer: =>
+    if @el.find(".problem-timer").length
+      @$timer = @el.find('.problem-timer')
+      @$display = @el.find('.minutes-left')
+      @startTime = new Date()
+      @totalSeconds = parseInt(@$timer.data('totalSecondsLeft'), 10)
+      @secondsBeforeWarning = parseInt(@$timer.data('secondsBeforeWarning'), 10)
+      @submittedBeforeTimeExpired = false
+
+      # The timer is initially hidden by CSS 
+      @$timer.show()
+
+      # Clear old timers, eg. if we press the submit button
+      if @timerId
+        clearInterval(@timerId)
+
+      # Sync every second
+      @timerId = setInterval(@syncTimer, 1000)
+
+      # Initialize and show timer
+      @syncTimer()
```

Comment:
The name of these values are somewhat of a misnomer. 

The backend knows the actual start time and uses it to calculate the total seconds left. So, if the person navigated to the back button and came back, the value of `data-total-seconds-left` would be less than what they started the exam with.

The `startTime` variable in this case refers to the last time the user has navigated to the page, not when the person first started the timed problem.